target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-wsq50.dts

```diff
@@ -0,0 +1,316 @@
+		};
+	};
+	bluetooth-enable {
+		gpio-hog;
+		gpios = <32 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+};
+
+&blsp1_uart1 {
+	pinctrl-0 = <&serial1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&blsp1_uart2 {
+	pinctrl-0 = <&serial2_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	bluetooth {
+		compatible = "csr,8811";
+
+		enable-gpios = <&tlmm 32 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&cryptobam {
+	status = "okay";
+};
+
+&blsp_dma {
+	status = "okay";
+};
+
+&gmac {
+	status = "okay";
+};
+
+&switch {
+	status = "okay";
+};
+
+&swport1 {
+	status = "disabled";
+
+	label = "lan4";
+};
+
+&swport2 {
+	status = "okay";
+
+	label = "lan3";
+};
+
+&swport3 {
+	status = "okay";
+
+	label = "lan2";
+};
+
+&swport4 {
+	status = "okay";
+
+	label = "lan1";
+};
+
+&swport5 {
+	status = "okay";
+};
+
+&wifi0 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "Zyxel-WSQ50";
+};
+
+&wifi1 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "Zyxel-WSQ50";
+};
+
+&pcie0 { /* QCA9984 */
+	status = "okay";
+	perst-gpio = <&tlmm 38 GPIO_ACTIVE_LOW>;
+	wake-gpio = <&tlmm 40 GPIO_ACTIVE_LOW>;
+	clkreq-gpio = <&tlmm 39 GPIO_ACTIVE_LOW>;
+
+	bridge@0,0 {
+		reg = <0x00000000 0 0 0 0>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		ranges;
+
+		wifi2: wifi@1,0 {
+			compatible = "qcom,ath10k";
+			reg = <0x00010000 0 0 0 0>;
+			qcom,ath10k-calibration-variant = "Zyxel-WSQ50";
+			qcom,cal-len = <0x2f20>;
+			qcom,cal-offset = <0x9000>;
+			qcom,mtd-name = "0:ART";
+		};
+	};
+};
+
+&usb2_hs_phy {
+	status = "okay";
+};
+
+&usb3_hs_phy {
+	status = "okay";
+};
+
+&usb3_ss_phy {
+	status = "okay";
+};
+
+&blsp1_i2c3 {
+	status = "okay";
+
+	tricolor: lp5562@30 {
+		clock-mode = [01];
+		compatible = "ti,lp5562";
+		reg = <0x30>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		status_red: chan@0 {
+			chan-name = "red:status";
+			// linux,default-trigger = "default-on";
+			led-cur = [c0];
+			max-cur = [ff];
+			reg = <0>;
+			color = <LED_COLOR_ID_RED>;
+		};
+
+		status_green: chan@1 {
+			chan-name = "green:status";
+			// linux,default-trigger = "default-on";
+			led-cur = [5c];
+			max-cur = [ff];
+			reg = <1>;
+			color = <LED_COLOR_ID_GREEN>;
+		};
+
+		status_blue: chan@2 {
+			chan-name = "blue:status";
+			// linux,default-trigger = "default-on";
+			led-cur = [ff];
+			max-cur = [ff];
+			reg = <2>;
+			color = <LED_COLOR_ID_BLUE>;
+		};
+	};
+};
+
+&blsp1_spi1 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor", "n25q128a11";
+		spi-max-frequency = <24000000>;
+		reg = <0x00>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition0@0 {
+				label = "0:QSEE";
+				reg = <0x00060000 0x00060000>;
+				read-only;
+			};
+			partition1@E0000 {
+				label = "u-boot";
+				reg = <0x000E0000 0x00080000>;
+				read-only;
+			};
+			partition2@160000 {
+				label = "env";
+				reg = <0x00160000 0x00010000>;
+				read-only;
```

Comment:
I assume this is U-Boot env so why is it read only?

Reply:
The label matches what is actually in the chip, the factory ROM kept it as read-only but I guess there's no reason to keep it as such.

Reply:
I am sorry, but `env` on its own is rather vague.

Is it the U-boot environment or?

Reply:
Yep it's the u-boot environment, I didn't know if u-boot itself depended on this, I'll try renaming it and see what happens

Reply:
Tried renaming this in the DTS file, however `cat /proc/mtd` still shows it as "env"

Reply:
Marking this as resolved - there's very little I can do about this

Reply:
Can you dump the live DTS?

I suspect you are hitting the same issue as other boards do, in so that the bootloader will patch in the nodes here.